src/update_engine/omaha_request_action.cc

```diff
@@ -498,6 +498,9 @@ bool OmahaRequestAction::ParsePackage(xmlDoc* doc,
   // propagate the urlBase vs packageName distinctions beyond this point.
   // From now on, we only need to use payload_urls.
   for (size_t i = 0; i < output_object->payload_urls.size(); i++) {
+    if (output_object->payload_urls[i].back() != '/') {
```

Comment:
Maybe you could add a comment that says that strings in payload_urls are guaranteed to be not empty, so you can safely use `back`?

Reply:
Good hint, I just added it to the condition instead of assuming that and crashing if they aren't